src/tls13.c

```diff
@@ -5779,7 +5779,9 @@ static int SendTls13CertificateVerify(WOLFSSL* ssl)
 #endif /* WOLFSSL_ASYNC_CRYPT */
 
     /* Final cleanup */
+#ifdef WOLFSSL_ASYNC_CRYPT
```

Comment:
Disabling this call will cause `args->sigData` to leak. I'd like to trace the root cause. What TLS 1.3 cipher suite and key type/size are you using for the MariaDB server?

Reply:
quite right on the leak. misread this.

Seems to be 8k certificates on the client/server side that doesn't occur on smaller sizes.

the server/client uses data from: https://github.com/MariaDB/server/tree/10.5/mysql-test/std_data

server:

key: server8k-key.pem
cert_file= server8k-cert.pem
ca_file=cacert.pem

client.key cert are:
client-cert.pem 
client-key.pem

client side is a gnutls based.

I tried reproducing this with
 examples/server/server -c /home/dan/repos/mariadb-server-10.4/mysql-test/std_data/server8k-cert.pem -k /home/dan/repos/mariadb-server-10.4/mysql-test/std_data/server8k-key.pem -A /home/dan/repos/mariadb-server-10.4/mysql-test/std_data/cacert.pem -i -d -v 4

however I was unable to generate the same results:

(from https://github.com/MariaDB/server/blob/10.4/mysql-test/main/ssl_8k_key.test)

appears to be RSA:

This is wolfssl from the v4.3.0-stable tag.

https://gist.github.com/grooverdan/cadb9499c2e3adc0bfbb0d22663155ff

failing that:
<pre>
git clone  --single-branch --branch 10.4-MDEV-22221-wolfssl-tls13 https://github.com/grooverdan/mariadb-server.git mariadb-server
cd mariadb-server
git submodule update --init --recursive
change mysql-test/main/ssl_8k_key.test to have TLSv1.3
mkdir  ../build
cd ../build
cmake -DPLUGIN_{ARCHIVE,TOKUDB,MROONGA,OQGRAPH,ROCKSDB,CONNECT,SPIDER}=NO   -DBUILD_CONFIG=mysql_release -DWITH_SSL=bundled  ../mariadb-server
make -j 8
mysql-test/mtr --mem {--manual-gdb} main.ssl_8k_key
</pre>